tests/lightning_fail_fast/run.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+#
+# Copyright 2020 PingCAP, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eux
+
+export GO_FAILPOINTS='github.com/pingcap/br/pkg/lightning/restore/SlowDownWriteRows=sleep(50);github.com/pingcap/br/pkg/lightning/restore/SetMinDeliverBytes=return(1)'
+
+for CFG in chunk engine; do
+  rm -f "$TEST_DIR/lightning-tidb.log"
+  run_sql 'DROP DATABASE IF EXISTS fail_fast;'
+
+  set +e
+  run_lightning --backend tidb --enable-checkpoint=0 --log-file "$TEST_DIR/lightning-tidb.log" --config "tests/$TEST_NAME/$CFG.toml"
+  ERRORCODE=$?
+  set -e
+
+  [ "$ERRORCODE" -ne 0 ]
```

Comment:
Why not?

```suggestion
  ! run_lightning --backend tidb --enable-checkpoint=0 --log-file "$TEST_DIR/lightning-tidb.log" --config "tests/$TEST_NAME/$CFG.toml"
```

Reply:
@YuJuncen i don't think it works.

```sh
#!/bin/bash
set -eu
true
echo 'should print this because `true`'
! false
echo 'should print this because `! false` == `true`'
! true
echo 'should not print this because `! true` == `false`'
false
echo 'should not print this because `false`'
```

```
should print this because `true`
should print this because `! false` == `true`
should not print this because `! true` == `false`
```

Reply:
@kennytm Well, bash is awesome. 🙃
So, we need to change those two `greps` below too, right?

Reply:
@YuJuncen no need to change the grep, the handling of error code is correct (note 0 = true, 1 = false):

```sh
#!/bin/bash
set -eu
true
echo $?, 'should print this because `true`'
! false
echo $?, 'should print this because `! false` == `true`'
! true
echo $?, 'should not print this because `! true` == `false`'
false
echo $?, 'should not print this because `false`'
```

```
0, should print this because `true`
0, should print this because `! false` == `true`
1, should not print this because `! true` == `false`
```

Reply:
@kennytm I mean these lines.

https://github.com/pingcap/br/blob/b5925e7ff7673b582fc29a6f18dc8f4d3108f386/tests/lightning_fail_fast/run.sh#L27-L28

Seems the return code isn't used here.

Reply:
oh. well yeah.